### ISSUE TITLE
feature: Improve all functions of the region

### ISSUE THREAD

**Issue by sjcsjc123** [ByteStorage/FlyDB#126]
| id     | issue     | people     |  process |
| -------- | -------- | -------- |   -------- | 
| 1 |Region method Put  | @sjcsjc123 | finish |
| 2 |Region method Get | @wangchenguang123  | finish|
| 3 |Region method Delete | @wangchenguang123  | finish|
| 4 |Region method GetStartKey |  @wangchenguang123 |finish |
| 5 |Region method GetEndKey | @wangchenguang123  |finish |
| 6 |Region method GetLeader | @wangchenguang123  | finish|
| 7 |Region method GetPeers | @wangchenguang123  | finish|
| 8 |Region method TransferLeader | @saeid-a |finish|
| 9 |Region method AddPeer | @saeid-a |finish |
| 10 |Region method RemovePeer | @saeid-a |finish |
| 11 |Region method GetSize | @wangchenguang123  | finish|

You can comment on the issue id you want to try below, and we will assign it to you

**Comment by sjcsjc123** [comment]
#131 is example

**Comment by wangchenguang123** [comment]
i want to have a try and i have a question How should I test the methods when i finsh code

**Comment by sjcsjc123** [comment]
Just write unit tests, the region section is mainly responsible for interacting with the underlying DB interface. If you want to try, please let me know if you want to try the issue id and I will assign it to you.

**Comment by wangchenguang123** [comment]
i want to try 2，3，4，5，6，7，11 and I am trying to understand 8,9,10

**Comment by sjcsjc123** [comment]
you can see https://docs.pingcap.com/zh/tidb/stable/tikv-overview to understand and give you suggestion.

**Comment by sjcsjc123** [comment]
2,3,4,5,6,7,11 now are for you

**Comment by saeid-a** [comment]
I want to try 8,9,10

**Comment by qishenonly** [comment]
> I want to try 8,9,10我想尝试8、9、10

OK，have a try.

**Comment by saeid-a** [comment]
Thanks, I want to clarify some things; when function `AddPeer` is run, am I correct in presuming that we should check that the `peer` does not already exist in `r.peers`? What error should we return in case of a duplicate peer?

**Comment by sjcsjc123** [comment]
> Thanks, I want to clarify some things; when function `AddPeer` is run, am I correct in presuming that we should check that the `peer` does not already exist in `r.peers`? What error should we return in case of a duplicate peer?

Yes, you need to check whether the peer exists. If so, return error.

**Comment by qishenonly** [comment]
> Thanks, I want to clarify some things; when function `AddPeer` is run, am I correct in presuming that we should check that the `peer` does not already exist in `r.peers`? What error should we return in case of a duplicate peer?

Your contributions are welcome.  We are very sorry that due to our mistake, we did not explain the submission specification to you in the early stage.  In future commits, the commit information of the commit needs to be followed by which problem was solved, if it is a single problem, such as issue156, then it needs to be followed by (#156), or if there are multiple tasks in the problem, such as this problem, then it needs to be followed by  (#126 ID 8). Thank you for your understanding.
